build-support/bin/packages.py

```diff
@@ -74,12 +74,95 @@ def owners(self) -> set[str]:
         return {row[1] for row in roles if row[0] == "Owner"}  # type: ignore[union-attr,index]
 
 
+def _pip_args(extra_pip_args: list[str]) -> tuple[str, ...]:
+    return (*extra_pip_args, "--quiet", "--no-cache-dir")
+
+
+def validate_pants_pkg(version: str, venv_dir: Path, extra_pip_args: list[str]) -> None:
+    subprocess.run(
+        [
+            venv_dir / "bin/pip",
+            "install",
+            *_pip_args(extra_pip_args),
+            f"pantsbuild.pants=={version}",
+        ],
+        check=True,
+    )
+
+    def run_venv_pants(args: list[str]) -> str:
```

Comment:
The positioning of this function within `validate_pants_pkg` was a bit of a confusing read to me given how large it is relative to outer function. Maybe either (1) move it to be the first item in the outer function (so the creation of the venv is with the remainder of the code in the function); or (2) move it out of this function and just take `venv_dir` as a parameter?